app/models/event.rb

```diff
@@ -30,6 +27,25 @@ def self.location_info(location)
     location.name
   end
 
+  def location
+    @location ||= begin
+      location_barcode = audit.record_data['location']
+      Location.find_by(barcode: location_barcode)
+    end
+  end
+
+  def coordinate
```

Comment:
Rubocop has already flagged this one that assigning nil is not needed as all ruby methods naturally return nil. I think you only need
```
@coordinate ||= begin 
if audit.id == labware.audits.last.id
 labware.coordinate
end
```

I presume the begin has a purpose but not clear on what it is.

Reply:
Ah fair enough thanks, will take the `nil` out

Reply:
Ah. Of course! You could use a setter if you wanted to do but it is fine as is.